modules/tensor_mechanics/src/materials/FiniteStrainCPSlipRateRes.C

```diff
@@ -0,0 +1,373 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "FiniteStrainCPSlipRateRes.h"
+
+template<>
+InputParameters validParams<FiniteStrainCPSlipRateRes>()
+{
+  InputParameters params = validParams<FiniteStrainCrystalPlasticity>();
+  return params;
+}
+
+FiniteStrainCPSlipRateRes::FiniteStrainCPSlipRateRes(const InputParameters & parameters) :
+    FiniteStrainCrystalPlasticity(parameters),
+    _resid(_nss,0.0),
+    _slip_rate(_nss,0.0),
+    _dsliprate_dgss(_nss,0.0),
+    _jacob(_nss),
+    _dsliprate_dsliprate(_nss)
+{
+ for (unsigned int i = 0;i < _nss; ++i)
+    _jacob[i].resize(_nss,0.0);
+
+ for (unsigned int i = 0;i < _nss; ++i)
```

Comment:
indentation is off for the two for loops, space after ```;``` and space after ```,``` everywhere